content/oci/oci_test.go

```diff
@@ -452,6 +452,86 @@ func TestStore_ContentBadPush(t *testing.T) {
 	}
 }
 
+func TestStore_ResolveByTagReturnsFullDescriptor(t *testing.T) {
+	content := []byte("hello world")
+	ref := "hello-world:0.0.1"
+	annotations := map[string]string{"name": "Hello"}
+	desc := ocispec.Descriptor{
+		MediaType:   "test",
+		Digest:      digest.FromBytes(content),
+		Size:        int64(len(content)),
+		Annotations: annotations,
+	}
+
+	tempDir := t.TempDir()
+	s, err := New(tempDir)
+	if err != nil {
+		t.Fatal("New() error =", err)
+	}
+	ctx := context.Background()
+
+	err = s.Push(ctx, desc, bytes.NewReader(content))
+	if err != nil {
+		t.Errorf("Store.Push() error = %v, wantErr %v", err, false)
+	}
+
+	err = s.Tag(ctx, desc, ref)
+	if err != nil {
+		t.Errorf("error tagging descriptor error = %v, wantErr %v", err, false)
+	}
+
+	resolvedDescr, err := s.Resolve(ctx, ref)
+	if err != nil {
+		t.Errorf("error resolving descriptor error = %v, wantErr %v", err, false)
+	}
+
+	if !reflect.DeepEqual(resolvedDescr, desc) {
+		t.Errorf("Store.Resolve() = %v, want %v", resolvedDescr, desc)
+	}
+
+	// descriptor resolved by tag should have annotations
+	if !reflect.DeepEqual(resolvedDescr.Annotations, annotations) {
+		t.Errorf("Store.Resolve() returned descriptor without annotations %v, want %v", resolvedDescr.Annotations, annotations)
+	}
+}
+
+func TestStore_ResolveByDigestReturnsPlainDescriptor(t *testing.T) {
+	content := []byte("hello world")
+	ref := "hello-world:0.0.1"
+	desc := ocispec.Descriptor{
+		MediaType:   "test",
+		Digest:      digest.FromBytes(content),
+		Size:        int64(len(content)),
+		Annotations: map[string]string{"name": "Hello"},
+	}
+
+	tempDir := t.TempDir()
+	s, err := New(tempDir)
+	if err != nil {
+		t.Fatal("New() error =", err)
+	}
+	ctx := context.Background()
+
+	err = s.Push(ctx, desc, bytes.NewReader(content))
+	if err != nil {
+		t.Errorf("Store.Push() error = %v, wantErr %v", err, false)
+	}
+
+	err = s.Tag(ctx, desc, ref)
+	if err != nil {
+		t.Errorf("error tagging descriptor error = %v, wantErr %v", err, false)
+	}
+
+	resolvedDescr, err := s.Resolve(ctx, string(desc.Digest))
+	if err != nil {
+		t.Errorf("error resolving descriptor error = %v, wantErr %v", err, false)
+	}
+
+	if reflect.DeepEqual(resolvedDescr, desc) {
```

Comment:
```suggestion
	if !reflect.DeepEqual(resolvedDescr, desc) {
```
?

Reply:
This is the scenario for resolving by digest so a plain descriptor is returned thus they will not be equal

Reply:
Oh right... I think it would be better to make a plain descriptor without annotation, and compare the resolved descriptor with the plain descriptor.

Reply:
updated